### PR TITLE
Make the Blacklight 8.x DocumentComponent less different

### PR DESCRIPTION
**Pull request by cbeer**
#2698 introduced some disruptive changes for downstream use. This smooths the migration path to 8.x a little better.

### CODE REVIEW

File: app/components/blacklight/document_component.rb
Comment (jcoyne) [inline app/components/blacklight/document_component.rb:82]: Should there be a conditional here?

File: app/views/catalog/_document.html.erb
Comment (jcoyne) [inline app/views/catalog/_document.html.erb:3]: Are you able to write test for this behavior?  It looks like it was inadvertently reverted because there was no test here before. Are you able to write one?

File: app/components/blacklight/document_component.rb
Comment (tpendragon) [inline app/components/blacklight/document_component.rb:77]: Can we add a note or example of what it takes to subclass, so we can explain the `**args` situation?

**Review by tpendragon — CHANGES_REQUESTED** [review]

**Comment by dkinzer** [comment]
Following up on yesterday's committer meeting. I volunteered to merge tjhis and do a beta6 release but there are two outstanding requests for changes here so I'm blocked until those are resolved.

**Review by tpendragon — APPROVED** [review]

**Comment by dkinzer** [comment]
spec failing with 
```  
NameError:
        undefined local variable or method `controller' for #<RSpec::ExampleGroups::BlacklightDocumentActionComponent::WithAPartialConfigured "render the partial" (./spec/components/blacklight/document/action_component_spec.rb:38)>
```

**Comment by jcoyne** [comment]
@dkinzer that's because of a new release of ViewComponent 3.  These test are failing on main: https://github.com/projectblacklight/blacklight/actions/runs/4225379473/jobs/7338207752

**Comment by dkinzer** [comment]
Thanks @jcoyne .   If this if failing on main too, than I don't see any more blockers.  I'm merging.

**Pull request merged**
